### PR TITLE
[Docs] Semantic Components

### PR DESCRIPTION
**Pull request by leecalcote**
**Notes for Reviewers**

- This PR enhances the description of Components to categorize them into semantically and non-semantically meaningful groups.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by miacycle — COMMENTED** [review]
I consider the term "semantic" a little difficult to digest at times, but after reading the description and difference between the two categories, I get it.

**Review by miacycle — APPROVED** [review]
Are the commented paragraphs to be filled-in later?

**Comment by leecalcote** [comment]
> Are the commented paragraphs to be filled-in later?

Thanks, @miacycle. yes, you're precisely correct. This is just a draft of what might become a further explanation. This particular section is borrowed from the description of semantically meaningful relationships.

File: docs/pages/concepts/logical/components.md [hunk 1]
```diff
@@ -9,7 +9,21 @@ list: include
...
+
+## Components as building blocks
+
+Meshery Models and their Components represent a standardized and reusable collection of building blocks used to describe both the desired and actual configuration of infrastructure pre and post-deployment. More than this Components are using to describe non-infrastructure concepts, for example, a component might capture a comment or a visual element that you've placed into a Meshery Design. To aid in distinguishing between types of Components, they are categorized into two groups: Semantic and Non-Semantic components. A component is considered semantically meaningful when it behaves as a direct representation of an aspect of your infrastructure. A component is considered non-semantically meaningful when it behaves as an annotation to aid in the comprehension of your designs.
```
Comment (winkletinkle) [inline docs/pages/concepts/logical/components.md:16]:
```suggestion
Meshery Models and their packaged Components represent a standardized and reusable collection of building blocks used to describe both the desired and actual configuration of infrastructure pre and post-deployment. More than this Components are using to describe non-infrastructure concepts, for example, a component might capture a comment or a visual element that you've placed into a Meshery Design. To aid in distinguishing between types of Components, they are categorized into two groups: Semantic and Non-Semantic components. A component is considered semantically meaningful when it behaves as a direct representation of an aspect of your infrastructure. A component is considered non-semantically meaningful when it behaves as an annotation to aid in the comprehension of your designs.
```

**Review by winkletinkle — APPROVED** [review]
I figure this note might help.

File: docs/pages/concepts/logical/components.md (hunk 1 above)
Comment (leecalcote) [reply docs/pages/concepts/logical/components.md:16]: This isn't a bad idea.

**Review by hamza-mohd — APPROVED** [review]

**Pull request merged**